account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,192 @@
+import { type ConnectionConfig, type SmartAccountSigner } from "@aa-sdk/core";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import type { Address } from "abitype";
+import {
+  type Hex,
+  type SignableMessage,
+  type TypedData,
+  type TypedDataDefinition,
+  hashMessage,
+  hashTypedData,
+} from "viem";
+import { TurnkeyClient, type TSignedRequest } from "@turnkey/http";
+import type { SignRawMessageMode, User } from "./client/types.js";
+import { buildStampedSignatureRequestBody } from "./client/base.js";
+import { AlchemySignerClient } from "./client/alchemy.js";
+
+interface ApiKeySignerClientParams {
+  orgId: string;
+  apiKey: {
+    publicKey: string;
+    privateKey: string;
+  };
+  connection: ConnectionConfig;
+}
+
+/**
+ * ApiKeySignerClient is a client that uses an API key to sign messages.
+ */
+class ApiKeySignerClient {
```

Comment:
This seems like a bunch of duplication of the logic that exists in the base signer client already? Does it not work for this to just extend that?

Reply:
there are a lot of abstract methods on `BaseSignerClient` that i don't need... should we just throw on some of them like we do in the react native client?

it almost feels like i need something i step lower than the `BaseSignerClient` b/c i don't care about a lot of methods it has involving auth, etc.

i'm not totally following why some of these methods are `BaseSignerClient` are even abstract to begin with, esp. the ones involving MFA.

Reply:
working on this on another branch for to compare

Reply:
will be extending it in an upcoming commit